bonfire/processor.py

```diff
@@ -117,6 +117,43 @@ def process_iqe_cji(
     return processed_template
 
 
+def process_reservation(
+    requester,
+    duration,
+    extension=False,
+    template_path=None,
+):
+    log.info("processing namespace reservation")
+
+    template_path = Path(template_path if template_path else conf.DEFAULT_RESERVATION_TEMPLATE)
+
+    if not template_path.exists():
+        raise FatalError("Reservation template file does not exist: %s", template_path)
+
+    with template_path.open() as fp:
+        template_data = yaml.safe_load(fp)
+
+    params = dict()
+    params["DURATION"] = duration
+    params["REQUESTER"] = requester
+
+    if extension:
+        res = get_reservation_by_requester(requester)
```

Comment:
This part probably needs to change if the way we extend reservations changes as discussed above